apps/sms/test/marionette/composer_test.js

```diff
@@ -42,6 +42,57 @@ marionette('Messages Composer', function() {
     );
   });
 
+  suite('Preserve message input while navigating', function() {
+    var composer, threadList, thread;
+    var message = 'test message';
+
+    function waitForThreadList() {
+      client.helper.waitForElement(threadList.mmsThread);
+    }
+
+    function createMMSThread() {
+      threadList.navigateToComposer();
+      messagesApp.addRecipient('a@b.c');
+      messagesApp.addRecipient('s@p.c');
+      composer.messageInput.sendKeys('MMS thread.');
+      messagesApp.send();
+      messagesApp.performHeaderAction();
+    }
+
+    setup(function() {
+      thread = messagesApp.Thread;
+      composer = messagesApp.Composer;
+      threadList = messagesApp.ThreadList;
+
+      client.contentScript.inject(
+        __dirname + '/mocks/mock_navigator_moz_icc_manager.js'
+      );
+
+      messagesApp.launch();
+      createMMSThread();
+      waitForThreadList();
+      threadList.mmsThread.tap();
+
+      composer.messageInput.tap();
+      composer.messageInput.sendKeys(message);
+    });
+
+    test('Message input is preserved when navigating to and from group-view',
+    function() {
+      thread.headerTitle.tap();
+      messagesApp.performHeaderAction();
```

Comment:
I'm really afraid that we can have intermittent failures here and below as we don't wait until group or report panel is displayed. We have in-progress patch that should separate headers headers for Thread/Report/Group panels. Unfortunately I can't check as composer tests are temporarily disabled on Treeherder :(

So can we have her something like this, before we perform header action?
```
client.helper.waitForElement(messagesApp.Participants.main); // for group view
client.helper.waitForElement(messagesApp.Report.main); // for report view
```